### PR TITLE
Introducing a Tracker phase-2 geometry based on T25 but compatible with new OT/BTL boundary

### PR DESCRIPTION
**Pull request by sroychow**
This the work of @emiglior @adewit 
#### PR description:
This PR introduces 

-  introducing a baseline geometry with TBPX layer1 featuring 3D sensors instead of planar sensors
-  new version of Pixel3DDigitizerAlgorithm with an improved description of the response of the 3D sensors before and after irradiation

The Tracker geometry used in the baseline CMS geometry for phase-2 is T24. On the other hand, T25 represents a more correct description of the actual layout of the detector chosen for phase-2, as it features 3D sensors in TBPX layer1. Unfortunately, T25 cannot be matched as it is with the latest developments of the MTD geometry (I16) as PR #39592  modified the boundary between OT and MTD. Note that Tracker T31 geometry is the same as T24 geometry, but with this new boundary.
Similarly, we propose to introduce a T32 geometry which is the same as T25 but with the new OT/MTD boundary and to adopt it as the new baseline phase-2 geometry so that it can be regularly exercised with RelVals and extended physics studies so that effects related to the change in the geometry can be decoupled wrt effects due to the change in the digitizer.
**NOTE**: T32 is not yet the ultimate geometry, as the actual geometry features two separate 3D sensors for each TBPX layer1 modules. The implementation of this geometry is less trivial as it requires a change of the detID scheme and will be available only later this year.

Link to the talk in simulation meeting :- https://indico.cern.ch/event/1247289/#48-phase-2-tracker-geometry-up

#### PR validation:
New wf introduced ```24834.0``` runs without errors 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
NA

### CODE REVIEW

**Comment by sroychow** [comment]
please test workflow 24834.0

**Comment by mmusich** [comment]
[test2026Geometry](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-80686e/30276/unitTests/src/Configuration/Geometry/test/test2026Geometry/testing.log) is failing with report:

```
===== Test "test2026Geometry" ====
GeometryExtended2026D98Reco_cff.py differs
GeometryDD4hepExtended2026D98Reco_cff.py differs

---> test test2026Geometry had ERRORS
TestTime:1
```

**Comment by sroychow** [comment]
please test with 24834.0

**Comment by tvami** [comment]
@cmsbuild , please test workflow 24834.0

**Comment by tvami** [comment]
> please test with 24834.0

"with" is with other PRs

**Comment by emiglior** [comment]
The error in the test unit is due to this difference

< from Geometry.CaloEventSetup.HGCalV9Topology_cfi import *
---
> from Geometry.CaloEventSetup.HGCalTopology_cfi import *

in the files

Configuration/Geometry/python/GeometryExtended2026D98Reco_cff.py
Configuration/Geometry/python/GeometryExtendedDDhep2026D98Reco_cff.py

I would like to ask advice on how to proceed.
I can just commit the version without "V9" but I am not sure that it is the right thing to do.

**Comment by emiglior** [comment]
The error in the test unit is due to this difference

`from Geometry.CaloEventSetup.HGCalV9Topology_cfi import *`

vs.

`from Geometry.CaloEventSetup.HGCalTopology_cfi import *`

in the files

Configuration/Geometry/python/GeometryExtended2026D98Reco_cff.py
Configuration/Geometry/python/GeometryExtendedDDhep2026D98Reco_cff.py

I would like to ask advice on how to proceed.
I can just commit the version without "V9" but I am not sure that it is the right thing to do.

**Comment by bsunanda** [comment]
I think use of HGCalTopology_cfi will be right. The one with v9 is removed.
________________________________

**Comment by sroychow** [comment]
@cmsbuild please test workflow 24834.0

**Comment by mmusich** [comment]
@cmsbuild please abort

File: Configuration/Geometry/cmsExtendedGeometry2026D98.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0"?>
```
Comment (mmusich) [inline Configuration/Geometry/cmsExtendedGeometry2026D98.xml:1]: why is this file needed? Seems a copy of `Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2026D98.xml`

File: Configuration/Geometry/cmsExtendedGeometry2026D98XML_cfi.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (mmusich) [inline Configuration/Geometry/cmsExtendedGeometry2026D98XML_cfi.py:1]: why is this file needed? 
seems a copy of `Geometry/CMSCommonData/python/cmsExtendedGeometry2026D98XML_cfi.py`

**Comment by mmusich** [comment]
> Found files with invalid states:

you might want to squash together 16252dc3a87cc145c3b7caef89b1ddcac6acb6ea and 3af57482312394e7691ed9dfd60dc5d6f50832b1 to avoid this.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by srimanob** [comment]
Hi @sroychow 

I just realize that I miss a comment on geometry. You compose the geometry with C18 which is the version of HGCAL that we don't use at the moment. So the relvals may contain something that we don't understand. The cleaning way is to make relvals based on `D95` with update on `T32`. You can make 2 versions (one with C17 and another with C18) in the PR also.

**Comment by sroychow** [comment]
Thanks @srimanob, following up on this.

**Comment by emiglior** [comment]
> Hi @sroychow
> 
> I just realize that I miss a comment on geometry. You compose the geometry with C18 which is the version of HGCAL that we don't use at the moment. So the relvals may contain something that we don't understand. The cleaning way is to make relvals based on `D95` with update on `T32`. You can make 2 versions (one with C17 and another with C18) in the PR also.

Hi @srimanob, what is the suggested option?

a/ two geometries
D98 = T32+C18+M10+I16+O9+F6
D99 = T32+C17+M10+I16+O9+F6

b/ replacing C18 with C17 in D98
D98 = T32+C17+M10+I16+O9+F6

**Comment by srimanob** [comment]
Hi @emiglior 
I think two geometries are good options, so you don't need to crate them later. Could you please switch D98 with C17, and D99 with C18, so they will be in order. Thanks.

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 3]
```diff
@@ -2383,6 +2387,20 @@ def condition(self, fragment, stepList, key, hasHarvest):
...
+        'HLTmenu': '@fake2',
+        'GT' : 'auto:phase2_realistic_T25',
+        'Era' : 'Phase2C17I13M9',
+        'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal', 'ALCAPhase2'],
+    },
+    '2026D99' : {
+        'Geom' : 'Extended2026D99',
+        'HLTmenu': '@fake2',
+        'GT' : 'auto:phase2_realistic_T25',
+        'Era' : 'Phase2C17I13M9',
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: Shouldn't this be `Phase2C18I13M9` ?
https://github.com/cms-sw/cmssw/blob/db4ff9a6cf2fea967f3d80ca084d195e6f5a17ed/Configuration/Eras/python/Era_Phase2C18I13M9_cff.py#L6

Comment (emiglior) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: I was cloning what it is done in upgradeWorkflowComponents.py for D95 (C17) and D96 (C18), but I am not expert on this

    '2026D95' : {
        'Geom' : 'Extended2026D95',
        'HLTmenu': '@fake2',
        'GT' : 'auto:phase2_realistic_T21',
        'Era' : 'Phase2C17I13M9',
        'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal', 'ALCAPhase2'],
    },
    '2026D96' : {
        'Geom' : 'Extended2026D96',
        'HLTmenu': '@fake2',
        'GT' : 'auto:phase2_realistic_T21',
        'Era' : 'Phase2C17I13M9',
        'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal', 'ALCAPhase2'],
    },

Comment (mmusich) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: Apparently `Phase2C18I13M9` is used only for D94 (=T24+C20+M10+I15+O9+F8), which looks confusing...

Comment (emiglior) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: is there any action needed on my side?

Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: Hi,
`Phase2C18I13M9` is used with geometry with HFNose only, so D94 (C20). Another with C17, C18 will use `Phase2C17I13M9`. This PR is good, and we can review the era later. Thanks for the comment.

**Comment by sroychow** [comment]
please test

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py (hunk 3 above)
Comment (mmusich) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: @srimanob can you clarify why the `Phase2C18I13M9`  is used with geometry with  C20 but not C18? Shouldn't it be called `Phase2C20I13M9` instead?

Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2401]: Hi @mmusich 
I agree. That is what I said to be follow up later, with @bsunanda

**Comment by srimanob** [comment]
@sroychow 
Should the test be done to both 24834.0 and 25234.0? Thx.

**Comment by mmusich** [comment]
test parameters:
   * workflow = 24834.0, 25234.0

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
+Upgrade

New D98 and D99 workflows run.

**Comment by bsunanda** [comment]
+geometry

**Comment by sroychow** [comment]
> Hi @sroychow Will this PR be the PR to merge? Should you move it out from a draft? Thx.

@srimanob done

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@bbilin, @kskovpen, @sunilUIET Please approve this PR. Like it to be integrated quickly

**Comment by kskovpen** [comment]
+pdmv

**Comment by bsunanda** [comment]
@perrotta @rappoccio Could you approve this PR and integrate this to the next IB?

**Comment by perrotta** [comment]
+1

**Comment by sroychow** [comment]
Link to the talk in simulation meeting :- https://indico.cern.ch/event/1247289/#48-phase-2-tracker-geometry-up
